src/decode/lzma.rs

```diff
@@ -225,7 +225,11 @@ where
         rangecoder: &mut rangecoder::RangeDecoder<'a, R>,
     ) -> error::Result<()> {
         loop {
-            if self.unpacked_size.is_some() && rangecoder.is_finished_ok()? {
+            if let Some(unpacked_size) = self.unpacked_size {
+                if self.output.len() as u64 >= unpacked_size {
```

Comment:
I'm wondering why the check isn't for equality. A larger output should trigger an error at this point instead.

Reply:
We're just doing `break` here. If it's an equality and the length goes above the expected value, it may take quite some time before the loop ends. With `>=` the program would break immediately and there's still a check later:
```rust
if let Some(len) = self.unpacked_size {
    if self.output.len() as u64 != len {
        return Err(error::Error::LZMAError(format!(
            "Expected unpacked size of {} but decompressed to {}",
            len,
            self.output.len()
        )));
    }
}
```
Is it correct or am I missing something?

Reply:
Makes sense given the check right after the loop.